src/noit_check.c

```diff
@@ -1451,12 +1471,45 @@ noit_poller_free_check_internal(noit_check_t *checker, mtev_boolean has_lock) {
 }
 void
 noit_poller_free_check(noit_check_t *checker) {
+  checker->flags |= NP_KILLED;
   noit_poller_free_check_internal(checker, mtev_false);
 }
-static int
-check_recycle_bin_processor(eventer_t e, int mask, void *closure,
-                            struct timeval *now) {
+static void
+check_recycle_bin_processor_internal() {
+  mtev_hash_iter iter = MTEV_HASH_ITER_ZERO;
   struct _checker_rcb *prev = NULL, *curr = NULL;
+  mtevL(noit_debug, "Scanning checks for cluster sync\n");
+  pthread_mutex_lock(&polls_lock);
+  while(mtev_hash_adv(&polls, &iter)) {
+    noit_check_t *check = (noit_check_t *)iter.value.ptr;
+    if(NOIT_CHECK_DELETED(check) && !noit_cluster_checkid_replication_pending(check->checkid)) {
+      char xpath[1024], idstr[UUID_STR_LEN+1];
+      mtev_uuid_unparse_lower(check->checkid, idstr);
+
+      mtevL(noit_debug, "cluster delete of %s complete.\n", idstr);
+      /* Set the config_seq to zero so it can be truly descheduled */
+      check->config_seq = 0;
+      noit_poller_deschedule(check->checkid, mtev_true);
```

Comment:
- config_seq is no longer monotonically increasing
- noit_poller_deschedule function is overloaded
cf. comments at noit_poller_deschedule implementation.